### ISSUE TITLE
spack-install     raised PermissionError

### ISSUE THREAD

**Issue by Mark-E-Hamilton** [LLNL/spack#38827]
### Steps to reproduce

I cloned spack and ran the example commands shown below.

host 998% source share/spack/setup-env.sh
host 999% spack install hdf5

### Error message

==> Error: cannot bootstrap any of the patchelf executables from spec "patchelf@0.13.1:0.13 %gcc target=x86_64" due to the following failures:
github-actions-v0.4 raised RuntimeError: The binary index is empty
github-actions-v0.3 raised RuntimeError: The binary index is empty
spack-install raised PermissionError: [Errno 13] Permission denied: '/usr/netpub/bin/bin'
Run `spack --debug ...` for more detailed errors

### Information on your system

I checked the directory reported in the error message. It's permissions were set such that only the owner could read it:

host 1125% ls -ld /usr/netpub/bin/bin
drwx--S--- 2 rvierte wg-eshpc-netpub 4096 Sep 22  2016 /usr/netpub/bin/bin

When the owner removed this directory the error went away. 

host 1021% spack debug report
* **Spack:** 0.20.0
* **Python:** 3.6.8
* **Platform:** linux-rhel7-cascadelake
* **Concretizer:** clingo
hpws194 1022% 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by psakievich** [comment]
I saw this today on my laptop.  It was related to the SNL proxy.  If I disconnected from the VPN and unset proxy vars then I bootstrapped right away.  Not super helpful for machines on the network though.

**Comment by alalazo** [comment]
Is there a way this can be reproduced outside SNL network?

**Comment by psakievich** [comment]
@alalazo @becker33 has seen this on the LLNL network as well.  IIRC the quick and dirty fix is `spack -k install` but I don't think it is something we can endorse long term.

**Comment by psakievich** [comment]
@alalazo I'm sorry. I'm not reading things super closely it seems.  Many new users at SNL have had trouble bootstrapping so I didn't look close enough at this issue.  I was getting an issue with empty bootstrap caches due to the network proxy, but I don't see an issue with the `/usr/netpub/bin/bin` permissions.  This seems to be something else.  So I think I was conflating the two issues.

**Comment by psakievich** [comment]
I guess it is not clear to me why the `/usr/netpub/bin/bin/` directory was getting used. @Mark-E-Hamilton did you have it in `PATH`?

Or perhaps it is related to the python instance you were using?
I'm wondering if this is more of a configuration error than a bug.

**Comment by Mark-E-Hamilton** [comment]
@alalazo @psakievich  I did not have /usr/netpub/bin/bin in my PATH. I presumed that the code was doing some sort of search in that directory, but I had a co-worker recreate that directory with the same permissions, cleaned out my entire spack clone in case any cached stuff interfered, and tried it again and this time it did not fail. I can't explain it, but go ahead and close this since it can't be reproduced. If it happens again (and I expect it will not) I'll let you know.
